.github/workflows/beam_IODatastoresCredentialsRotation.yml

```diff
@@ -0,0 +1,78 @@
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Rotate IO-Datastores Cluster Credentials
+
+on:
+  schedule:
+    - cron: '0 2 1 * *'
+  workflow_dispatch:
+
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: none
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.issue.number || github.sha || github.head_ref || github.ref }}-${{ github.event.schedule || github.event.sender.login }}'
+  cancel-in-progress: true
+
+env:
+  GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_ACCESS_TOKEN }}
+  GRADLE_ENTERPRISE_CACHE_USERNAME: ${{ secrets.GE_CACHE_USERNAME }}
+  GRADLE_ENTERPRISE_CACHE_PASSWORD: ${{ secrets.GE_CACHE_PASSWORD }}
+
+jobs:
+  beam_IODatastoresCredentialsRotation:
+    if: |
+      github.event_name == 'workflow_dispatch' ||
+      github.event_name == 'schedule'
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    timeout-minutes: 100
+    name: ${{ matrix.job_name }}
+    strategy:
+      matrix:
+        job_name: ["beam_IODatastoresCredentialsRotation"]
+        job_phrase: ["N/A"]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup repository
+        uses: ./.github/actions/setup-action
+        with:
+          comment_phrase: ${{ matrix.job_phrase }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          github_job: ${{ matrix.job_name }}
+      - name: Starting credential rotation
+        run: |
+          gcloud container clusters update io-datastores --start-credential-rotation --zone=us-central1-a --quiet
+      - name: Rebuilding the nodes
+        run: |
+          gcloud container clusters upgrade io-datastores --node-pool=pool-1 --zone=us-central1-a --quiet
+      - name: Completing the rotation
+        run: |
+          gcloud container clusters update io-datastores --complete-credential-rotation --zone=us-central1-a --quiet
+# TODO: Send email to dev@beam.apache.org if something went wrong during credentials rotation
```

Comment:
Based on the conversation with @damccorm we agreed to omit this for now

Reply:
I think this is actually a slightly different case since its going to dev@beam.apache.org instead of builds@beam.apache.org. I think we should keep emails to the dev list, especially something like this that has a high probability of indicating something that we need to address immediately.

Reply:
We are going to implement the `Send email` step in the separate PR.

Reply:
Sounds good.